### ISSUE TITLE
About the evaluation time

### ISSUE THREAD

**Issue by wyzjack** [mbzuai-oryx/Video-ChatGPT#29]
Hi authors,

Thanks for the great work! I am running your evaluation code on ActivityQA dataset following [https://github.com/mbzuai-oryx/Video-ChatGPT/blob/main/quantitative_evaluation/README.md](url) and the inference time takes up to 4-5 hours on a single A100 80G GPU. I am wondering whether that is normal? Thanks.

I would appreciate it very much if you could reply.

Thanks

**Comment by hanoonaR** [comment]
Hi @wyzjack,

Thanks for your interest in our work and for reaching out with your question.

Indeed, the ActivityNet-QA dataset is fairly large, containing around 8000 questions, which can make the inference process quite time-consuming, especially when run on a single GPU.

In our work, to expedite the [inference](https://github.com/mbzuai-oryx/Video-ChatGPT/blob/main/video_chatgpt/eval/run_inference_activitynet_qa.py), we divided the workload across multiple GPUs. We accomplished this by a simple hack and splitting the contents of the JSON file into smaller chunks, each of which was processed on a separate GPU. After the inference process, we combined the results from each GPU. In our experience, running the task on 4 or 8 GPUs can substantially decrease the inference time.

I hope this information is helpful. Let me know if you have any other questions!
